homeassistant/components/sensor/netatmo.py

```diff
@@ -0,0 +1,115 @@
+"""
+homeassistant.components.sensor.netatmo
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+This sensor connects to the cloud service provided by netatmo and uses
+the api.
+To successfully connect it needs the client_id and client_secret from
+a netatmo developer account and your username and password from the
+netatmo user account
+"""
+from datetime import timedelta
+import logging
+import requests
+
+from homeassistant.const import (STATE_UNKNOWN)
+from homeassistant.util import Throttle
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import TEMP_CELCIUS
+
+SENSOR_TYPES = {
+    'Temperature': [TEMP_CELCIUS],
+    'Humidity': ['%'],
+    'CO2': ['ppm'],
+    'Noise': ['dB'],
+    'Pressure': ['mb']
+}
+
+sensor_url = "https://api.netatmo.com//api/getstationsdata?access_token="
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'NETAMTO Sensor'
+DEFAULT_METHOD = 'GET'
+
+# Return cached results if last scan was less then this time ago
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=300)
+
+
+# pylint: disable=unused-variable
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Gets a list of the available stations and sets up the sensors. """
+
+    client_id = config.get('client_id', None)
+    client_secret = config.get('client_secret', None)
+    username = config.get('username', None)
+    password = config.get('password', None)
+    dev = list()
+    request = requests.post("https://api.netatmo.net/oauth2/token",
+                            data={'grant_type': 'password',
+                                  'client_id': client_id,
+                                  'client_secret': client_secret,
+                                  'username': username,
+                                  'password': password})
+    access_token = request.json()["access_token"]
+    request = requests.get(sensor_url + access_token).json()
```

Comment:
I think that all requests should have a timeout (maybe `timeout=10` like the `rest` sensor).